packages/trpc/server/routers/loggedInViewer/deleteCredential.handler.ts

```diff
@@ -151,8 +180,7 @@ export const deleteCredentialHandler = async ({ ctx, input }: DeleteCredentialOp
               metadata: {
                 ...metadata,
                 apps: {
-                  ...metadata?.apps,
-                  [appSlug]: undefined,
```

Comment:
I am unable to catch the bug we are fixing here by using `removeAppFromEventTypeMetadata`

Reply:
Also, if we still want to use a fn for some reason, I think we should make a generic one, something like `removePropFromObject`

Reply:
On the event types page we parse the app metadata. If one of the apps is undefined it throws a zod error and stops the loading of the event types page completely.

Reply:
That is weird. This change is really old. Maybe something change elsewhere recently. 

I am okay with this change for now.